llvm/lib/Object/DXContainer.cpp

```diff
@@ -288,6 +290,51 @@ Error DirectX::RootSignature::parse(StringRef Data) {
                             llvm::Twine(FValue));
   Flags = FValue;
 
+  Current = Begin + RootParametersOffset;
```

Comment:
Should we just validate that `Current == Begin + RootParametersOffset` here? Are there legitimate cases where that might not be true?

Reply:
Looks like this was modified to just assert and assume that the root parameters immediately follow the header. 

I'm not 100% convinced that's the right thing to do - if the file format allows for an arbitrary RootParametersOffset then I don't think we should assume that any arbitrary data we read in is tightly packed.

Reply:
Looking at D3D's root signature deserializer, it _doesn't_ make the assumption that the root parameters immediately follow the header.  If we make this assumption then this will mean that there are root signature blobs that D3D can parse successfully that this code will fall over on.

Reply:
@bogner - I'm going to resolve this conversation, I think it's been addressed now.